### DISCUSSION TITLE
Why is not the MultiUploadEventArguments object a list of UploadEventArguments?

### DISCUSSION THREAD

**Discussion by JS-Aibel** [zauberzeug/nicegui#3752] [Ideas / Feature Requests]
### Question

The MultiUploadEventArguments object has lists for contents, names, and types, where an uploaded file corresponds to the same index in each list. Why are the properties for each file not packed together in the same element?

**Comment by falkoschindler** [comment]
Hi @JS-Aibel,

I'm not sure if there was a specific reason for it other than that "the same element" would require another data structure like `tuple`, `dict`, `NamedTuple`, `dataclass`, ... But yes, a list of "files" might have been a better choice than multiple lists.

Changing the data structure would introduce a breaking change. But we _could_ extend it with an additional field `files: List[UploadFile]`. I'll convert this discussion into a feature request in case we want to further discuss this API.
